index.html

```diff
@@ -16,28 +16,51 @@ <h2>Try it:</h2>
       <textarea maxlength="1024" id="input-text" cols="60" rows="10">aeiou</textarea>
       <br>
       <button type="submit">Play</button>
+      <a href="javascript:stopPlayback()"><button type="button">Stop</button></a>
+      <a href="javascript:exportAudio()"><button type="button">Export</button></a>
     </form>
     <audio id="result-audio">
       Your browser does not support the HTML5 audio element.
     </audio>
     <script type="text/javascript">
       let lastText = null;
       let audio = null;
+      let playing = false;
       function tts() {
         let text = document.getElementById('input-text').value;
 
         if (text !== lastText) {
           audio = new Audio(`/tts?text=${encodeURIComponent(text)}`);
           audio.addEventListener('error', () => {
+            playing = false;
             lastText = null;
           });
           audio.addEventListener('playing', () => {
+            playing = true;
             lastText = text;
           });
+          audio.addEventListener('ended', () => {
+            playing = false;
+          });
         }
 
         if (audio) audio.play();
       }
+
+      function stopPlayback() {
+        audio.pause();
+        audio.currentTime = 0;
+        playing = false;
+      }
+
+      function exportAudio() {
+        let text = document.getElementById('input-text').value;
+        window.location = `/tts?text=${encodeURIComponent(text)}`;
```

Comment:
I assume this replaces the current page right?  It might be nice if it opened in a new tab or otherwise triggered a download so that clicking export doesn't navigate you away from the current page.

Reply:
On Firefox, it leaves you on the same page and just opens the download prompt. Haven't tested it on other browsers but I imagine it does the same thing.

Reply:
If that doesn't cut it, this will open the URL in a new tab so there's no way the user gets redirected off the main page. Your choice
```javascript
function exportAudio() {
  let text = document.getElementById('input-text').value;
  window.open(`/tts?text=${encodeURIComponent(text)}`, '_blank').focus();  
}
```

Reply:
I confirmed that in Chromium it replaces the current page (I assume this is because by default Firefox treats .wav files as downloads whereas by default Chromium plays them inline).

I decided to just make the download link a normal <a> tag, which allows the user to click it, right click save as, copy the link and paste it into some other downloading tool, etc.